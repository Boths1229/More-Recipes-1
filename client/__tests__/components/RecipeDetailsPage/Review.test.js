import React from 'react';
import { shallow } from 'enzyme';
import Review from '../../../components/RecipeDetailsPage/Reviews';
import MockData from '../../__mocks__/actions/recipes';

let props;

const setup = () => {
  props = {
    reviews: MockData.recipeData.reviews,
    handleSubmit: () => {},
    handleChange: () => {},
    onFocus: () => {},
    reviewError: '',
  };

  return shallow(<Review {...props} />);
};

describe('Review Component', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper.find('button').text()).toBe('Submit');
  });

  it('should render message if recipe have not been reviewed', () => {
    const wrapper = setup();
    wrapper.setProps({ reviews: [] });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h6').text()).toBe('This recipe have not been reviewed');
  });

  it('should render error if error exists', () => {
    const wrapper = setup();
    wrapper.setProps({ reviewError: 'Review cannot be Empty' });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').at(4).text()).toBe('Review cannot be Empty');
  });
});
