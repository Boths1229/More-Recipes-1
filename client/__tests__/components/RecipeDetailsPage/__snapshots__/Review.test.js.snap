// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Component should render correctly 1`] = `
<section
  className="border-recipes"
>
  <h4
    className="section-reviews centered"
    style={
      Object {
        "backgroundColor": "#2b3034",
      }
    }
  >
    Reviews
  </h4>
  <ul
    style={
      Object {
        "maxHeight": "300px",
        "overflowY": "scroll",
      }
    }
  >
    <li
      className="review-list"
      key="0"
    >
      <div>
        <div>
          <div
            className="header"
          >
            <strong
              className="primary-font username"
            >
              sannikay
            </strong>
            <strong
              className="primary-font pull-right"
              style={
                Object {
                  "paddingRight": "10px",
                }
              }
            >
              7 days ago
            </strong>
          </div>
          <p>
            hi there

          </p>
          <hr />
        </div>
      </div>
    </li>
  </ul>
  <form
    id="review-form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="comment"
      >
        Post a review
      </label>
      <textarea
        className="form-control"
        id="comment"
        name="review"
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        rows="5"
      />
    </div>
    <button
      className="btn vote-button pull-right"
      type="submit"
    >
      Submit
    </button>
  </form>
</section>
`;

exports[`Review Component should render error if error exists 1`] = `
<section
  className="border-recipes"
>
  <h4
    className="section-reviews centered"
    style={
      Object {
        "backgroundColor": "#2b3034",
      }
    }
  >
    Reviews
  </h4>
  <ul
    style={
      Object {
        "maxHeight": "300px",
        "overflowY": "scroll",
      }
    }
  >
    <li
      className="review-list"
      key="0"
    >
      <div>
        <div>
          <div
            className="header"
          >
            <strong
              className="primary-font username"
            >
              sannikay
            </strong>
            <strong
              className="primary-font pull-right"
              style={
                Object {
                  "paddingRight": "10px",
                }
              }
            >
              7 days ago
            </strong>
          </div>
          <p>
            hi there

          </p>
          <hr />
        </div>
      </div>
    </li>
  </ul>
  <form
    id="review-form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="comment"
      >
        Post a review
      </label>
      <textarea
        className="form-control"
        id="comment"
        name="review"
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        rows="5"
      />
      <div
        className="alert alert-danger"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Review cannot be Empty
      </div>
    </div>
    <button
      className="btn vote-button pull-right"
      type="submit"
    >
      Submit
    </button>
  </form>
</section>
`;

exports[`Review Component should render message if recipe have not been reviewed 1`] = `
<section
  className="border-recipes"
>
  <h4
    className="section-reviews centered"
    style={
      Object {
        "backgroundColor": "#2b3034",
      }
    }
  >
    Reviews
  </h4>
  <h6
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    This recipe have not been reviewed
  </h6>
  <form
    id="review-form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="comment"
      >
        Post a review
      </label>
      <textarea
        className="form-control"
        id="comment"
        name="review"
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        rows="5"
      />
    </div>
    <button
      className="btn vote-button pull-right"
      type="submit"
    >
      Submit
    </button>
  </form>
</section>
`;
